osf/models/registrations.py

```diff
@@ -290,6 +293,50 @@ def date_withdrawn(self):
     def withdrawal_justification(self):
         return getattr(self.root.retraction, 'justification', None)
 
+    @property
+    def provider_specific_metadata(self):
+        """Surfaces the additional_metadata fields supported by the provider.
+
+        Also formats the reults to inherit any additional field descriptors defined
+        by the provider and to simplify consumpption by the APIt.
+        """
+        additional_metadata = self.additional_metadata or {}
+
+        if not self.provider or not self.provider.additional_metadata_fields:
+            return []
+
+        provider_supported_metadata = []
+        for field_desc in self.provider.additional_metadata_fields:
+            metadata_field = {
+                'field_value': additional_metadata.get(field_desc['field_name'], '')
+            }
+            metadata_field.update(field_desc)
+            provider_supported_metadata.append(metadata_field)
+
+        return provider_supported_metadata
+
+    def update_provider_specific_metadata(self, values):
+        """Updates additional_metadata fields supported by the provider.
+
+        Fields listed in values that are not supported by the current provider will be ignored.
+        Fields supported by the provider that are not listed in values will not be altered.
+        """
+        if not self.provider or not self.provider.additional_metadata_fields:
+            return
+
+        if not self.additional_metadata:
+            self.additional_metadata = {}
+
+        provider_supported_fields = {
+            entry['field_name'] for entry in self.provider.additional_metadata_fields
+        }
+        for entry in values:
+            key = entry['field_name']
+            value = entry['field_value']
+            if key in provider_supported_fields:
```

Comment:
it's just a noop if the field isn't supported? Might be confusing if adding to the wrong provider.

Reply:
Yeah, this is very much a judgement call that I waffled between (exception vs silent fail). I'm open to 400-ing if folks agree that that's best. FWIW, FE doesn't provide a path to send unsupported fields, so this would only impact folks using the API directly...in which case maybe 400 is clearer?

Reply:
400 is just generally what we do when sent bad json. I'd argue this is just better for integrators, good errors messages are better then good docs.

Reply:
400 it is